plugins/utils/indent-blankline.nix

```diff
@@ -5,87 +5,48 @@
   pkgs,
   ...
 }:
-with lib; let
-  basePluginPath = ["plugins" "indent-blankline"];
-in {
-  # TODO: Those warnings were introduced on 2023/10/17.
-  # Please, remove in early December 2023.
-  imports =
-    # Removed options
-    (
-      map
-      (
-        optionName:
-          mkRemovedOptionModule
-          (basePluginPath ++ [optionName])
-          ''
-            Please use the new options.
-            See https://github.com/lukas-reineke/indent-blankline.nvim.
-          ''
-      )
-      [
-        "charBlankline"
-        "charList"
-        "charListBlankline"
-        "charHighlightList"
-        "spaceCharBlankline"
-        "spaceCharHighlightList"
-        "spaceCharBlanklineHighlightList"
-        "useTreesitter"
-        "indentLevel"
-        "maxIndentIncrease"
-        "showFirstIndentLevel"
-        "showEndOfLine"
-        "showFoldtext"
-        "disableWithNolist"
-        "filetype"
-        "bufnameExclude"
-        "strictTabs"
-        "showCurrentContextStartOnCurrentLine"
-        "contextCharBlankline"
-        "contextCharListBlankline"
-        "charPriority"
-        "contextStartPriority"
-        "contextPatterns"
-        "useTreesitterScope"
-        "contextPatternHighlight"
-        "disableWarningMessage"
-      ]
-    )
-    # New options
-    ++ (
-      mapAttrsToList
-      (
-        oldName: newPath:
-          mkRenamedOptionModule
-          (basePluginPath ++ [oldName])
-          (basePluginPath ++ newPath)
-      )
-      {
-        char = ["indent" "char"];
-        showCurrentContext = ["scope" "enabled"];
-        showTrailingBlanklineIndent = ["whitespace" "removeBlanklineTrail"];
-        filetypeExclude = ["exclude" "filetypes"];
-        buftypeExclude = ["exclude" "buftypes"];
-        showCurrentContextStart = ["scope" "showStart"];
-        contextChar = ["scope" "char"];
-        contextCharList = ["scope" "char"];
-        contextHighlightList = ["scope" "highlight"];
-      }
-    );
-
-  options.plugins.indent-blankline =
-    helpers.neovim-plugin.extraOptionsOptions
-    // {
-      enable = mkEnableOption "indent-blankline.nvim";
-
-      package = helpers.mkPackageOption "indent-blankline" pkgs.vimPlugins.indent-blankline-nvim;
-
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "indent-blankline";
+    originalName = "indent-blankline.nvim";
+    luaName = "ibl";
+    defaultPackage = pkgs.vimPlugins.indent-blankline-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    # TODO introduced 2024-03-10: remove 2024-05-10
+    deprecateExtraOptions = true;
+    optionsRenamedToSettings = [
+      "debounce"
+      "viewportBuffer"
```

Comment:
does this work with  the suboptions (viewportBuffer.min, viewportBuffer.max) ?

Reply:
The warning is triggered for the root option.
Maybe it is better to specify for each of the suboptions still.

Reply:
I'd find this more re-assuring  (and more helpful too)

Reply:
I agree, done.